openoa/utils/downloader.py

```diff
@@ -0,0 +1,425 @@
+    Args:
+        record_id(:obj:`int`): the Zenodo record id
+        outfile_path(:obj:`str`): path to save files to
+
+    Returns:
+        Files saved to the asset data folder:
+         1. record_details.json, which details the zenodo api details
+         2. all files available for the record_id
+    """
+    
+    
+    url_zenodo = r"https://zenodo.org/api/records/"
+
+    record_id = str(record_id)
+    
+    r = requests.get(url_zenodo + record_id)
+    
+    r_json = r.json()
+    
+    
+    logger.info("======")
+    logger.info("Title: " + r_json["metadata"]["title"])
+    logger.info("Version: " + r_json["metadata"]["version"])
+    logger.info("URL: " + r_json["links"]["latest_html"])
+    logger.info("Record DOI: " + r_json["doi"])
+    logger.info("License: " + r_json["metadata"]["license"]["id"])
+    logger.info("======\n")
+    
+       
+    # create outfile_path if it does not exist
+    if not os.path.exists(outfile_path):
+        os.makedirs(outfile_path)
+    
+    
+    # save record details to json file
+    outfile = outfile_path+"record_details.json"
+    
+    with open(outfile, "wb") as f:
+        f.write(r.content)
+
+        
+    # download all files
+    files = r_json["files"]
+    for f in files:
+        
+        url_file = f["links"]["self"]
+        
+        file_name = f["key"]
+                
+        outfile = outfile_path+file_name
+        
+        
+        # check if file exists
+        if os.path.exists(outfile):
+            
+            
+            # if it does check the checksum is correct
+            with open(outfile, "rb") as f_check:
+                file_hash = hashlib.md5()
+                while chunk := f_check.read(8192):
+                    file_hash.update(chunk)
+        
+            if f["checksum"][4:]==file_hash.hexdigest():
+                logger.info("File already exists: " + file_name)
+            
+            
+            # download and unzip if the checksum isn't correct
+            else:
+                
+                logger.info("Downloading: " + file_name)
+                logger.info("File size: " + str(round(f["size"]/(1024*1024),2)) + "MB")       
+
+                download_file(url_file,outfile)
+
+                logger.info("Saved to: " + str(outfile) + "\n")
+
+                if outfile.endswith(".zip"):
+                    with ZipFile(outfile) as zipfile:
+                        zipfile.extractall(outfile_path)
+        
+        
+        # download and unzip if the file doesn't exist
+        else:
+            
+            logger.info("\nDownloading: " + file_name)
+            logger.info("File size: " + str(round(f["size"]/(1024*1024),2)) + "MB")       
+
+            download_file(url_file,outfile)
+
+            logger.info("Saved to: " + str(outfile) + "\n")
+
+            if outfile.endswith(".zip"):
+                with ZipFile(outfile) as zipfile:
+                    zipfile.extractall(outfile_path)
+
+
+
+def get_era5(
```

Comment:
Both this and the MERRA2 method should be renamed to `get_monthly_xx` to be clear about what data a user should be expecting.

Reply:
I agree with Rob. I modified these functions locally to download hourly data (albeit very slowly) and would like to contribute get_hourly_era5 (and merra2) functions to the module after this PR is merged. Also, I'm really glad you contributed these monthly reanalysis downloading functions, which were relatively easy to modify to download hourly data!

Reply:
Done, although I've gone with get_xxx_monthly, rather than get_monthly_xxx. My thinking being to group together any MERRA2 and ERA5 functions together. No particularly strong opinions though.

Reply:
I'm totally fine with your alternative, and I think it reads more cleanly anyway.